include/modem/location.h

```diff
@@ -161,14 +161,56 @@ struct location_datetime {
 struct location_data_details_gnss {
 	/** Number of satellites tracked at the time of event. */
 	uint8_t satellites_tracked;
+	/** Number of satellites used for a fix. */
+	uint8_t satellites_used;
 	/** PVT data. */
 	struct nrf_modem_gnss_pvt_data_frame pvt_data;
+	/**
+	 * GNSS search time in milliseconds.
+	 *
+	 * This is the time since last start of GNSS operation to the fix or timeout
+	 * including any time LTE has blocked GNSS.
+	 *
+	 * @ref pvt_data member has @ref nrf_modem_gnss_pvt_data_frame.execution_time member
+	 * which indicates cumulative GNSS execution time since last start
+	 * excluding any time LTE has blocked GNSS.
+	 */
+	uint32_t search_time;
 };
 
-/** Location details. */
+/** Location details for cellular. */
+struct location_data_details_cellular {
+	/** Number of neighbor cells. */
+	uint8_t ncells_count;
+	/** Number of GCI (surrounding) cells. */
+	uint8_t gci_cells_count;
+};
+
+/** Location details for Wi-Fi. */
+struct location_data_details_wifi {
+	/** Number of Wi-Fi APs. */
+	uint16_t ap_count;
+};
+
+/**
+ * Location details.
+ *
+ * Only one of the child structures is filled most of the time depending on the method
+ * found in @ref location_event_data.method member of the parent structure.
+ * There are cases when cellular and Wi-Fi data are combined into a single cloud request
+ * in which case data details for both methods are filled.
```

Comment:
Is there any way to determine when cellular and Wi-Fi have been combined, I guess not?

Reply:
Not from the event data, except knowing the config you used for `location_request()`, i.e., if cellular and Wi-Fi were next to each other in the method list. So the app calling `location_request()` knows it but Memfault monitoring only the events won't know it with the current API.

Reply:
There's now special treatment for the method fields of `LOCATION_EVT_STARTED` and `LOCATION_EVT_FALLBACK` when combined method is used.
`LOCATION_EVT_STARTED` has it in the current NCS/main and that's a bug I didn't realize earlier.

Please check the way those evens are composed, and understand that when both Wi-Fi and cellular are used, we do report just one of them in the events as follows:

- `LOCATION_EVT_STARTED`: The first method in the given config
- `LOCATION_EVT_LOCATION`: If accuracy > 100m, use cellular. Otherwise Wi-Fi.
- `LOCATION_EVT_TIMEOUT`/`LOCATION_EVT_ERROR`: Wi-Fi
- `LOCATION_EVT_FALLBACK`: Whichever is the first one on the list.

Note: Above applies when we would set method to `LOCATION_METHOD_INTERNAL_WIFI_CELLULAR`, which is not in public API. It doesn't apply if methods were not combined due to conditions mentioned in the API.

Reply:
I've added a commit on how it would look like if there was a combined LOCATION_METHOD_WIFI_CELLULAR method. This simplifies many things in the implementation. We can discuss and see if we keep this or not.
You cannot give this method in the method list for `location_request()`.